apps/src/templates/census2017/YourSchool.jsx

```diff
@@ -27,25 +30,58 @@ const styles = {
   }
 };
 
-export default class YourSchool extends React.Component {
+export default class YourSchool extends Component {
   static propTypes = {
-    alertHeading: React.PropTypes.string,
-    alertText: React.PropTypes.string,
-    alertUrl: React.PropTypes.string,
-    hideMap: React.PropTypes.bool
+    alertHeading: PropTypes.string,
+    alertText: PropTypes.string,
+    alertUrl: PropTypes.string,
+    hideMap: PropTypes.bool
   };
 
+  constructor(props) {
+    super(props);
+    this.responsive = new Responsive();
+    this.state = {
+      windowWidth: $(window).width(),
+      windowHeight: $(window).height(),
+      mobileLayout: this.responsive.isResponsiveCategoryInactive('md')
+    };
+  }
+
   componentDidMount() {
     if (!this.props.hideMap) {
       $('#map').appendTo(ReactDOM.findDOMNode(this.refs.map)).show();
     }
+    // Resize handler.
+    window.addEventListener('resize', _.debounce(this.onResize, 100).bind(this));
+  }
+
+  onResize() {
+    const windowWidth = $(window).width();
+    const windowHeight = $(window).height();
+
+    // We fire window resize events when the grippy is dragged so that non-React
+    // controlled components are able to rerender the editor. If width/height
+    // didn't change, we don't need to do anything else here
+    if (windowWidth === this.state.windowWidth &&
+        windowHeight === this.state.windowHeight) {
+      return;
+    }
+
+    this.setState({
+      windowWidth: $(window).width(),
+      windowHeight: $(window).height()
+    });
+
+    this.setState({mobileLayout: this.responsive.isResponsiveCategoryInactive('md')});
   }
 
   render() {
+    const desktop = (this.responsive.isResponsiveCategoryActive('lg') || this.responsive.isResponsiveCategoryActive('md'));
```

Comment:
Because `isResponsiveCategoryActive` simply checks to see whether we're currently bigger than that category's width, it should be sufficient to just check against 'md' rather than both sizes.